### PR TITLE
[Feature] Add C++ API for SDK

### PR DESCRIPTION
**Pull request by lzhangzz**
- mmcls
  - [x] classification
- mmdet
  - [x] detection
  - [x] instance segmentation
- mmseg
  - [x] segmentation
- mmedit
  - [x] restoration
- mmocr
   - [x] text detection
   - [x] text recognition
- mmpose
  - [x] pose detection
- mmrotate
  - [x] rotated detection

### CODE REVIEW

**Comment by lvhan028** [comment]
Please fix linting issue

**Comment by lvhan028** [comment]
Build demo failed with `MMDEPLOY_BUILD_SDK_MONOLITHIC`  and `MMDEPLOY_BUILD_EXAMPLES`

The error log shows:
```
[ 91%] Building CXX object demo/csrc/CMakeFiles/pose_detector.dir/pose_detector.cxx.o
cd /home/PJLAB/lvhan/Documents/projects/open-mmlab/mmdeploy/build/demo/csrc && /usr/bin/g++-7  -I/home/PJLAB/lvhan/Documents/projects/open-mmlab/mmdeploy/csrc/mmdeploy/apis/c/mmdeploy/.. -isystem /home/PJLAB/lvhan/Documents/projects/thirdparty/opencv-4.5.5/install/include/opencv4 -O3 -DNDEBUG -std=gnu++1z -o CMakeFiles/pose_detector.dir/pose_detector.cxx.o -c /home/PJLAB/lvhan/Documents/projects/open-mmlab/mmdeploy/demo/csrc/pose_detector.cxx
/home/PJLAB/lvhan/Documents/projects/open-mmlab/mmdeploy/demo/csrc/pose_detector.cxx:2:10: fatal error: mmdeploy/pose_detector.hpp: No such file or directory
 #include "mmdeploy/pose_detector.hpp"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by irexyc** [comment]
forget to install c++ examples, but may be fixed by #810

File: csrc/mmdeploy/apis/cxx/mmdeploy/common.hpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        return MMDEPLOY_PIXEL_FORMAT_COUNT;
+    }
+  }
+#endif
+ private:
+  mmdeploy_mat_t desc_;
+};
+
+template <typename T>
+class Result_ {
```
Comment (lvhan028) [inline csrc/mmdeploy/apis/cxx/mmdeploy/common.hpp:118]: Cool

**Review by lvhan028 — APPROVED** [review]

**Review by irexyc — APPROVED** [review]

**Pull request merged**